tests/rptest/tests/test_si_cache_space_leak.py

```diff
@@ -0,0 +1,143 @@
+    """
+    The test checks that SI cache doesn't exhibit a resource leak.
+    In order to do this the test puts pressure to SI cache by settings its
+    size to the minimum value. Then it uses FranzGoVerifiableProducer(Consumer)
+    to produce and consume data via SI. The retention on the topic has to be
+    small enough in order for SI to be involved. The test checks that no segment
+    files are opened in the cache directory.
+    """
+
+    topics = (TopicSpec(partition_count=100, replication_factor=3), )
+    test_defaults = {
+        'default':
+        dict(log_segment_size=1024 * 1024,
+             cloud_storage_cache_size=10 * 1024 * 1024),
+    }
+
+    def __init__(self, test_context, *args, **kwargs):
+        test_name = test_context.test_name
+        si_params = self.test_defaults.get(
+            test_name) or self.test_defaults.get('default')
+        si_settings = SISettings(**si_params)
+        self._segment_size = si_params['log_segment_size']
+        extra_rp_conf = {
+            'disable_metrics': True,
+            'election_timeout_ms': 5000,
+            'raft_heartbeat_interval_ms': 500,
+            'segment_fallocation_step': 0x1000,
+            'cloud_storage_enable_remote_read': True,
+            'cloud_storage_enable_remote_write': True,
+            'retention_bytes': self._segment_size,
+        }
+        super().__init__(test_context,
+                         num_brokers=3,
+                         extra_rp_conf=extra_rp_conf,
+                         si_settings=si_settings)
+        self._ctx = test_context
+        self._verifier_node = test_context.cluster.alloc(
+            ClusterSpec.simple_linux(1))[0]
+        self.logger.info(
+            f"Verifier node name: {self._verifier_node.name}, segment_size: {self._segment_size}"
+        )
+
+    def init_producer(self, msg_size, num_messages):
+        self._producer = FranzGoVerifiableProducer(self._ctx, self.redpanda,
+                                                   self.topic, msg_size,
+                                                   num_messages,
+                                                   [self._verifier_node])
+
+    def init_consumer(self, msg_size, num_messages, concurrency):
+        self._consumer = FranzGoVerifiableRandomConsumer(
+            self._ctx, self.redpanda, self.topic, msg_size, num_messages,
+            concurrency, [self._verifier_node])
+
+    def free_nodes(self):
+        super().free_nodes()
+        wait_until(lambda: self.redpanda.sockets_clear(self._verifier_node),
+                   timeout_sec=120,
+                   backoff_sec=10)
+        self.test_context.cluster.free_single(self._verifier_node)
+
+    def validate_disk_space(self):
+        pass
+
+    @cluster(num_nodes=4)
+    @matrix(message_size=[10000],
+            num_messages=[100000],
+            num_read=[1000],
+            concurrency=[2])
+    def test_si_cache(self, message_size, num_messages, num_read, concurrency):
```

Comment:
question: afaik these parameters will only get one value each here. is the use of matrix for adding more test values in future?

Reply:
the only reason is to be able to easily add new variants in the future